RenderSystems/GLSupport/src/GLX/OgreGLXWindow.cpp

```diff
@@ -601,8 +601,10 @@ namespace Ogre
             }
             else if( _glXSwapIntervalMESA )
                 _glXSwapIntervalMESA( vsync ? mVSyncInterval : 0 );
-            else
+            else if( _glXSwapIntervalSGI )
                 _glXSwapIntervalSGI( vsync ? mVSyncInterval : 0 );
+            else
+                throw std::runtime_error("no glx swap interval function found");
```

Comment:
same

Reply:
here you can make the calls conditional as ´!mIsExternalGLControl && vsync`

Reply:
Do you mean the whole block? I wouldn't have thought that was ok since not calling the function is not the same as calling it with `0` right? What if vsync was on but was then turned off? Maybe that situation wouldn't happen, idk. I'm happy to do what ever you suggest, but can you clarify if you mean the entire block starting with `if (! mIsExternalGLControl )` or a narrower scope, something like in the final else case:

```c++
if (_glXSwapIntervalSGI) {
  _glXSwapIntervalSGI( vsync ? mVSyncInterval : 0 );
} else if (vsync) {
  // throw exception because vsync requested and _glXSwapIntervalSGI not found
}  // implicit else, pass on missing function because no vsync requested
```

Reply:
Edited the previous comment to say "not ok" rather than "ok"...

Reply:
Sorry reading this again, that sentence makes no sense. What I meant to say was that I did not think it would be acceptable to not call one of the swap interface functions when vsync was false, because I assumed not calling one of those functions and calling one of those functions with argument `0` would have different outcomes.